src/sentry/search/events/filter.py

```diff
@@ -764,3 +768,130 @@ def format_search_filter(term, params):
             conditions.append(converted_filter)
 
     return conditions, projects_to_filter, group_ids
+
+
+class QueryFilter(QueryBase):
+    """ Filter logic for a snql query """
+
+    def resolve_where(self, query: str) -> None:
+        try:
+            parsed_terms = parse_search_query(query, allow_boolean=True, params=self.params)
+        except ParseError as e:
+            raise InvalidSearchQuery(f"Parse error: {e.expr.name} (column {e.column():d})")
+
+        for term in parsed_terms:
+            if isinstance(term, SearchFilter):
+                conditions = self.format_search_filter(term)
+                if conditions:
+                    self.where.append(conditions)
+
+    def resolve_params(self) -> None:
+        """Keys included as url params take precedent if same key is included in search
+        They are also considered safe and to have had access rules applied unlike conditions
+        from the query string.
+        """
+        # start/end are required so that we can run a query in a reasonable amount of time
+        if "start" not in self.params or "end" not in self.params:
+            raise InvalidSearchQuery("Cannot query without a valid date range")
+
+        self.where.append(Condition(self.column("timestamp"), Op.GTE, self.params["start"]))
```

Comment:
These params should all be validated. `start` and `end` should be datetimes, `project_id` should be ints etc.